### PR TITLE
Crude squash and rebase of the #14096

### PR DESCRIPTION
**Pull request by XVilka**
Due to the size of the PR and multiple merged I might be wrong in some places. So please do a careful review.

cc @a1ext @mrexodia @GustavoLCR @Maijin 

I didn't address your comments yet, lets make it work first.

A gentle rebase of https://github.com/radare/radare2/pull/14096

### CODE REVIEW

**Comment by a1ext** [comment]
@XVilka seems like `w32.c` is not being compiled, make sure it is in meson.build

**Comment by Maijin** [comment]
in the PR `libr/debug/p/native/w32.c                       // !! not used by anything else` maybe it's not compiled on purpose?

**Comment by a1ext** [comment]
Then move `is_pe_hdr` to an another .c :)

вс, 9 июня 2019 г. в 19:14, Maijin <notifications@github.com>:

**Comment by a1ext** [comment]
@XVilka have you changed anything in the code? because hers PR last commit was ok ([appveyor](https://ci.appveyor.com/project/radare/radare2-shvdd/builds/25014484/job/r8fvu0yt768vg8pj))

**Comment by XVilka** [comment]
@a1ext of course I changed (and a lot), because while she was working on that PR the underlying code changed as well. So I have to synchronise/adapt the diffs.

**Comment by radare** [comment]
But the debugger works as expected now?

**Comment by a1ext** [comment]
> But the debugger works as expected now?
> […](#)
> On 9 Jun 2019, at 20:03, Anton Kochkov ***@***.***> wrote: @a1ext of course I changed (and a lot), because while she was working on that PR the underlying code changed as well. So I have to synchronise/adapt the diffs. — You are receiving this because your review was requested. Reply to this email directly, view it on GitHub, or mute the thread.

it doesn't compile, take a look at appveyor

File: libr/debug/p/debug_native.c [hunk 1]
```diff
@@ -1425,8 +1349,7 @@ static RList *r_debug_native_modules_get (RDebug *dbg) {
 	if (list && !r_list_empty (list)) {
 		return list;
 	}
-#endif
-#if __WINDOWS__
+#elif  __WINDOWS__
 	list = r_w32_dbg_modules (dbg);
```
Comment (GustavoLCR) [inline libr/debug/p/debug_native.c]: `r_w32_dbg_modules` is not defined so the pointer is getting truncated, same on line 1201

**Comment by XVilka** [comment]
It compiles now, but for some reason the build still fails:
```
[XX] WAT DO db\formats\mangling\bin
Cannot spawn children with code 3221225477
Error: Cannot spawn children with code 3221225477
    at ChildProcess.<anonymous> (C:\projects\radare2-shvdd\radare2-regressions\new\node_modules\r2pipe\index.js:280:12)
    at ChildProcess.emit (events.js:189:13)
    at maybeClose (internal/child_process.js:970:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
[XX] WAT DO db\formats\mangling\mangling
Cannot spawn children with code 3221225477
```

**Comment by radare** [comment]
Can you download this bin and try executing it in local? I guess theres a null deref or stgh that makes it crash

**Comment by GustavoLCR** [comment]
@XVilka This patch fixes the crash. Debugger still not working though
[patch.txt](https://github.com/radare/radare2/files/3273722/patch.txt)

**Comment by XVilka** [comment]
@GustavoLCR by the way, you can send PRs to another PRs - just sending PR to my branch should work. FYI.

**Comment by radare** [comment]
So... moving for 3.7? one week left r2-3.6 and this PR still looks like not working and pretty untested

**Comment by XVilka** [comment]
Yes, it is unfinished and untested.

**Comment by radare** [comment]
@GustavoLCR can you have a look at this PR?

**Comment by XVilka** [comment]
@GustavoLCR it crashes now

**Comment by XVilka** [comment]
@GustavoLCR what is the status of this PR? Is it ready to be merged?

**Comment by GustavoLCR** [comment]
@XVilka I think it should be ok to merge now, and get people testing to see if there is anything broken

**Comment by XVilka** [comment]
I squashed everything into one commit, so it would be easier to find the problem with bisecting if required.

**Pull request merged**